drivers/tee/tee_shm_pool.c

```diff
@@ -28,6 +28,7 @@ static int pool_op_gen_alloc(struct tee_shm_pool_mgr *poolm,
 	va = gen_pool_alloc(genpool, s);
 	if (!va)
 		return -ENOMEM;
+	memset(va, 0, s);
```

Comment:
Surely this must result in a warning since `va` is an `unsigned long` and not a pointer.